Revamp of the data modeling tools page with KB resources added to it (#428)
neo4j/docs-getting-started: modules/ROOT/pages/data-modeling/data-modeling-tools.adoc

:description: See a list of tools that you can use when designing a data model.
:page-role: labs-label

//Check Mark
:check-mark: icon:check[]

//Cross Mark
:cross-mark: icon:times[]

Neo4j offers a range of no-code tools to get you started with data modeling.
Find here the available options.

== Neo4j Data Importer

image::data-importer-1.png[600,600,role="popup-link"]

Besides being a tool for xref:data-import/index.adoc[data importing], you can use Neo4j Data Importer to sketch out a graph model and map your input data to its structure and properties.
You can access it:

* Via the *Import* tab in link:https://neo4j.com/product/auradb/?ref=docs-nav-get-started[Neo4j Aura].
* As a standalone version link:https://data-importer.neo4j.io/[with secure connection only] or link:https://data-importer.graphapp.io/[with both secure and insecure connections].

For more information, refer to the link:{docs-home}/data-importer/current/modeling[Neo4j Data Importer -> Data modeling].

== Arrows.app

image::northwind-graph-model.png[600,600,role="popup-link"]

link:https://arrows.app[Arrows.app] is a no-code visualization platform which allows whiteboarding ideas into a graph model.

> It is ideal for making a sketch of a graph, especially if you want to design a domain model for your data.

Reviewer: ```suggestion
It is ideal for designing a domain model for your data.
```

With this platform, you can:

* Draft your own whiteboard from scratch or import data from JSON files and plain text.
* Create, modify, and delete nodes and relationships with their labels and properties without writing any code.
* Export the visualization model as Cypher statements, so you can load the same data model into a Neo4j database.

[role=label--labs]
== Cypher Workbench

image::cypher-workbench.png[600,600,role="popup-link"]

link:https://help.neo4j.solutions/neo4j-solutions/cypher-workbench/[Cypher Workbench] is a cloud-based tool that assists Neo4j developers in creating and maintaining solutions built on top of Neo4j. 
It combines no-code visual solutions as the ones available in link:https://arrows.app[Arrows.app] while also offering importing options similar to link:{docs-home}/data-importer[Neo4j Data Importer].

With this platform, you can:

> * Create a data model from scratch or import data from from JSON files.

Author: I think both, from what is written here https://help.neo4j.solutions/neo4j-solutions/cypher-workbench/model_import/

* Reverse-engineer data models from existing Neo4j databases.
* Use Cypher statements to augment the current data model, including node labels, relationship types, and properties.
* Validate your model (naming conventions, constraints, data, common mistakes, etc).
* Use a business scenarios tool for capturing questions and scenarios of use cases.

> * Import scenarios from Excel, Google Sheets, or plain text.

Author: Apparently it's a specific resource in the platform, an option for data modeling https://help.neo4j.solutions/neo4j-solutions/cypher-workbench/scenarios_model/

Maybe this would be more correct or at least generic?
```suggestion
* Import data from Excel, Google Sheets, or plain text.
```

To install it:

> * Check out the code from the link:https://github.com/neo4j-labs/cypher-workbench[GitHub Repository].

Author: Maybe we should just link to this page: https://help.neo4j.solutions/neo4j-solutions/cypher-workbench/installation/

* Ensure that you have Node.js on your machine.
* Follow the instructions in the Cypher Workbench project README or refer to the link:https://help.neo4j.solutions/neo4j-solutions/cypher-workbench/[documentation].

== Other tools

> Here is a list of other non-Neo4j tools that can be used for data modeling:

Reviewer: ```suggestion
There are other non-Neo4j tools that can be used for data modeling:
```

* link:https://mermaid.live/[Mermaid]: general data modeling tool (not specifically for graph databases), based on Markdown.
Ideal for documenting modeling strategies.
* link:https://plantuml.com/[PlantUML]: application for creating diagrams from plain text.

> Works better with version control rather than first model design.

Reviewer: ```suggestion
This is more for version control than model design.
```

* link:https://hackolade.com/[Hackolade]: a tool to design, document, and communicate data models and schemas.

> Built to support data modeling of Neo4j node labels and relationship types.

Reviewer: ```suggestion
Built to support the kind of data modeling specific to Neo4j with node labels and relationship types.
```

== Tools comparison

[options=header,cols="^.^2,^.^,^.^2,^.^2"]
|===
| Tool
| Free
| Import
| Export

| Data Importer
| {check-mark}
| .csv, .tsv
| -

| Arrows
| {check-mark}
| JSON
| Image, Cypher, JSON,URL, GraphQL

| Cypher Workbench
| {cross-mark}
| Cypher Workbench JSON, Apoc.meta.schema, Arrows JSON
| JSON

| PlantUML
| {check-mark}

> | PUML,JSON

Reviewer: ```suggestion
| PUML, JSON
```
